mailpile/auth.py

```diff
@@ -121,11 +121,10 @@ def _do_redirect(self):
         path = self.data.get('_path', [None])[0]
 
         # These are here to prevent people from abusing this to redirect to
-        # arbitrary URLs on the Internet. The first assertion blocks normal
-        # http:// and https:// URLs. The second is a catch-all that prevents
-        # any path that looks like it starts with a protocol: foo:/stuff/.
-        assert('://' not in path)
-        assert('/' in path.split(':')[0])
+        # arbitrary URLs on the Internet.
+        assert('://' not in path)          # https://, http://, ftp://, ...
+        assert(path[:2] != '//')           # //www.example.com/
+        assert('/' in path.split(':')[0])  # mailto:user@example.com
```

Comment:
IMHO, relying on *urlparse* does the same job (maybe more), is more explicit, robust & better reviewed :

```
from urlparse import urlparse

url = urlparse(path)
assert(not url.scheme and not url.netloc)
```

My too late cents ;-)